neuland/shared/R3BNeulandCommon.h

```diff
@@ -0,0 +1,106 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#ifndef NEULANDCOMMON_H
+#define NEULANDCOMMON_H
+
+#include <cmath>
+#include <limits>
+#include <vector>
+
+namespace Neuland
+{
+    bool GetJumps(double* data,
+                  unsigned int size,
+                  double_t threshold,
+                  unsigned int averageSize = 3,
+                  std::vector<unsigned int>* output = nullptr);
+
+    // Constants
+
+    constexpr auto __sqrt12 = 3.464101615;
+    constexpr auto NaN = std::numeric_limits<double>::quiet_NaN();
+    constexpr auto Inf = std::numeric_limits<double>::infinity();
+
+    // Usefull functions I do not want to redefine in every file
+
+    template <typename T>
+    constexpr T Sqr(const T val)
+    {
+        return val * val;
+    }
+
+    // NaN2Value maps NaNs and Infs to a finite value.
+    // Usefull if you want to fill a possible nan variable into a histogram
+    constexpr double NaN2Value(const double val, const double valIfNaN = 0.)
+    {
+        return (std::isfinite(val) ? val : valIfNaN);
+    }
+
+    // Initialize variables from Birk' s Law
+
+    constexpr double BirkdP = 1.032;
+    constexpr double BirkC1 = 0.013 / BirkdP;
+    constexpr double BirkC2 = 9.6e-6 / (BirkdP * BirkdP);
+
+    constexpr double GetLightYieldAfterBirk(const int charge, const double length_mm, const double edep_MeV)
+    {
+        return edep_MeV / (1. + BirkC1 * (Sqr(charge) > 1 ? 7.2 / 12.6 : 1.) * (edep_MeV / length_mm) +
```

Comment:
I would rather catch the charge == 0. and length_mm == 0. cases outside of this function here.
But in case you want to catch it here I will push the modified one.

Reply:
Catching those cases outside would leave the function nice and short as it is now, but inside would make the code more resilient and there is less that can go wrong. So I would appreciate that.

If there is time, someone (probably me) should have a look how often these cases happen in the simulation, and if it is a concern.

Reply:
Well I would say it leaves the user in a false state of safety, because the function will then catch some cases the user might want to treat differently or is even not aware of those cases.

Reply:
> not aware of those cases

That is more or less the point. There is / should be one way to handle these cases, and the problem is solved there. "Nothing to worry about, move along."

Not saying the current logic is the correct way to solve it.